src/components/Register/Register.jsx

```diff
@@ -34,9 +23,21 @@ const Register = () => {
     return errors;
   };
 
-  const handleRegisterSubmit = (event) => {
-    dispatch(registerUser(event));
-    navigate("/login");
+  const handleRegisterSubmit = async (event) => {
+    try {
+      const { user, error } = await supabase.auth.signUp({
+        email: event.email,
+        password: event.password,
+      });
+
+      if (error) {
+        throw new Error(error.message);
+      }
+
+      navigate("/login");
+    } catch (error) {
+      console.error("Failed to register user: ", error.message);
+    }
```

Comment:
Let's not do this in component and move it to actions.

Reply:
done